Plutarch/Api/V1.hs

```diff
@@ -95,68 +97,81 @@ import qualified PlutusTx.AssocMap as PlutusMap
-              '[ PBuiltinList (PAsData PTxInInfo)
-               , PBuiltinList (PAsData PTxOut)
-               , PValue
-               , PValue
-               , PBuiltinList (PAsData PDCert)
-               , PBuiltinList (PAsData (PTuple '[PStakingCredential, PInteger]))
-               , PPOSIXTimeRange
-               , PBuiltinList (PAsData PPubKeyHash)
-               , PBuiltinList (PAsData (PTuple '[PDatumHash, PDatum]))
-               , PTxId
+          ( PDataRecord
+              '[ "inputs" ':= PBuiltinList (PAsData PTxInInfo)
+               , "outputs" ':= PBuiltinList (PAsData PTxOut)
+               , "fee" ':= PValue
+               , "mint" ':= PValue
+               , "dcert" ':= PBuiltinList (PAsData PDCert)
+               , "wdrl" ':= PBuiltinList (PAsData (PTuple PStakingCredential PInteger))
+               , "validRange" ':= PPOSIXTimeRange
+               , "signatories" ':= PBuiltinList (PAsData PPubKeyHash)
+               , "data" ':= PBuiltinList (PAsData (PTuple PDatumHash PDatum))
+               , "id" ':= PTxId
                ]
           )
       )
   deriving
     (PMatch, PIsData)
     via PIsDataReprInstances PTxInfo
+  deriving (PDataFields) via (DerivePDataFields PTxInfo)
 
 instance PUnsafeLiftDecl PTxInfo where type PLifted PTxInfo = Plutus.TxInfo
 deriving via (DerivePConstantViaData Plutus.TxInfo PTxInfo) instance (PConstant Plutus.TxInfo)
 
 instance PIsDataRepr PTxInfo where
   type
     PIsDataReprRepr PTxInfo =
-      '[ '[ PBuiltinList (PAsData PTxInInfo)
-          , PBuiltinList (PAsData PTxOut)
-          , PValue
-          , PValue
-          , PBuiltinList (PAsData PDCert)
-          , PBuiltinList (PAsData (PTuple '[PStakingCredential, PInteger]))
-          , PPOSIXTimeRange
-          , PBuiltinList (PAsData PPubKeyHash)
-          , PBuiltinList (PAsData (PTuple '[PDatumHash, PDatum]))
-          , PTxId
+      '[ '[ "inputs" ':= PBuiltinList (PAsData PTxInInfo)
+          , "outputs" ':= PBuiltinList (PAsData PTxOut)
+          , "fee" ':= PValue
+          , "mint" ':= PValue
+          , "dcert" ':= PBuiltinList (PAsData PDCert)
+          , "wdrl" ':= PBuiltinList (PAsData (PTuple PStakingCredential PInteger))
+          , "validRange" ':= PPOSIXTimeRange
+          , "signatories" ':= PBuiltinList (PAsData PPubKeyHash)
+          , "data" ':= PBuiltinList (PAsData (PTuple PDatumHash PDatum))
+          , "id" ':= PTxId
           ]
        ]
   pmatchRepr dat f =
     (pmatchDataRepr dat) ((DRHCons (f . PTxInfo)) $ DRHNil)
 
 newtype PScriptContext (s :: S)
-  = PScriptContext (Term s (PDataList '[PTxInfo, PScriptPurpose]))
+  = PScriptContext
+      ( Term
+          s
+          ( PDataRecord
+              '[ "txInfo" ':= PTxInfo
+               , "purpose" ':= PScriptPurpose
+               ]
+          )
+      )
   deriving
     (PMatch, PIsData)
     via PIsDataReprInstances PScriptContext
+  deriving (PDataFields) via (DerivePDataFields PScriptContext)
```

Comment:
not sure what you mean

Reply:
Indeed, this isn't right. I thought this was `ScriptPurpose`.

Reply:
In fact, I added the comment in the wrong place! See a few lines below the instance for `PScriptPurpose`, it is bogus, because the type family doesn't reduce.